src/SmartHotel.Clients/SmartHotel.Clients/Services/IoT/RoomDevicesDataService.cs

```diff
@@ -45,16 +51,36 @@ public async Task<RoomTemperature> GetRoomTemperatureAsync()
 				}
 				else
 				{
-					//TODO: Handle the case where we haven't actually retrieved data yet
-					// simulates values returned for IoT device
-					var currentTemp = 70;
-					var desiredTemp = 76;
-					return new RoomTemperature( new SensorValue( currentTemp ), new SensorValue( desiredTemp ) );
+                    var roomData = await RoomSensorData(token, _roomId);
+
+                    // TODO: more generic approach to remove duplicated code in GetRoomAmbientLightAsync()
+                    RoomTemperature temperature = null;
```

Comment:
I was thinking more of an approach where the **StartCheckingRoomSensorData** makes a call right away if we don't have any data AND only if the endpoint is defined (I think I forgot that) and store off something we know we're still waiting for the result.

Then when we get to this logic (and the ambient light that's basically the same) we wait for the result to complete and then can just check **_currentSensorDataBySensorDataType** for the value. If it's not there something went wrong.

Not sure what is a available for us to use, but a SemaphoreSlim could work.

This way we only end up with a single "initialization" call to get the sensor data, whereas right now we'd most likely be making 3 calls